src/databricks/sqlalchemy/dialect/__init__.py

```diff
@@ -0,0 +1,300 @@
+          a list of column names that make up the primary key
+
+        name
+          optional name of the primary key constraint.
+
+        """
+        # TODO: implement this behaviour
+        return {"constrained_columns": []}
+
+    def get_foreign_keys(self, connection, table_name, schema=None, **kw):
+        """Return information about foreign_keys in `table_name`.
+
+        Given a :class:`_engine.Connection`, a string
+        `table_name`, and an optional string `schema`, return foreign
+        key information as a list of dicts with these keys:
+
+        name
+          the constraint's name
+
+        constrained_columns
+          a list of column names that make up the foreign key
+
+        referred_schema
+          the name of the referred schema
+
+        referred_table
+          the name of the referred table
+
+        referred_columns
+          a list of column names in the referred table that correspond to
+          constrained_columns
+        """
+        # TODO: Implement this behaviour
+        return []
+
+    def get_indexes(self, connection, table_name, schema=None, **kw):
+        """Return information about indexes in `table_name`.
+
+        Given a :class:`_engine.Connection`, a string
+        `table_name` and an optional string `schema`, return index
+        information as a list of dictionaries with these keys:
+
+        name
+          the index's name
+
+        column_names
+          list of column names in order
+
+        unique
+          boolean
+        """
+        # TODO: Implement this behaviour
+        return []
+
+    def get_table_names(self, connection, schema=None, **kwargs):
+        TABLE_NAME = 1
+        with self.get_driver_connection(
+            connection
+        )._dbapi_connection.dbapi_connection.cursor() as cur:
+            sql_str = "SHOW TABLES FROM {}".format(
+                ".".join([self.catalog, schema or self.schema])
+            )
+            data = cur.execute(sql_str).fetchall()
+            _tables = [i[TABLE_NAME] for i in data]
+
+        return _tables
+
+    def get_view_names(self, connection, schema=None, **kwargs):
+        VIEW_NAME = 1
+        with self.get_driver_connection(
+            connection
+        )._dbapi_connection.dbapi_connection.cursor() as cur:
+            sql_str = "SHOW VIEWS FROM {}".format(
+                ".".join([self.catalog, schema or self.schema])
+            )
+            data = cur.execute(sql_str).fetchall()
+            _tables = [i[VIEW_NAME] for i in data]
+
+        return _tables
+
+    def do_rollback(self, dbapi_connection):
+        # Databricks SQL Does not support transactions
+        pass
+
+    def has_table(self, connection, table_name, schema=None, **kwargs) -> bool:
+        """SQLAlchemy docstrings say dialect providers must implement this method"""
+
+        schema = schema or "default"
+
+        try:
+            res = connection.execute(f"DESCRIBE TABLE {table_name}")
```

Comment:
is this called frequently? Is this the best way to know if a table exists in Databricks?

Reply:
It's called _all the time_. Based on light experimentation this is the fastest way to check it right now. Using the built-in `.tables()` method of our cursor is actually quite slow in comparison.

Reply:
any replies from this? You could also use `show tables in catalog.schema like "name"`. But this is not blocking as long as it works